### PR TITLE
Feature/off chain external auth

### PR DESCRIPTION
**Pull request by demonsh**
Please review circuits/offchain/credentialAtomicQuerySigV2.circom

### CODE REVIEW

**Review by AndriianChestnykh — COMMENTED** [review]
The partial review for now.
Didn't check the testvectorsgen yet

File: circuits/lib/query/credentialJsonLDAtomicQueryMTP.circom [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+- claim schema
+- claim ownership and issuance state
+- claim non revocation state
+- claim expiration ?
+- query JSON-LD claaim's field
+
+IdOwnershipLevels - Merkle tree depth level for personal claims
+IssuerLevels - Merkle tree depth level for claims issued by the issuer
+ClaimLevels - Merkle tree depth level for claim JSON-LD document
+valueLevels - Number of elements in comparison array for in/notin operation if level =3 number of values for
```
[the remark this answers is not in the document]
Comment (AndriianChestnykh) [reply circuits/lib/query/credentialJsonLDAtomicQueryMTP.circom:25]: Should comment for the `valueArraySize`

File: circuits/offchain/credentialAtomicQuerySigOffChain.circom [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    */
+
+    // flag indicates if merkleized flag set in issuer claim (if set MTP is used to verify that
+    // claimPathValue and claimPathKey are stored in the merkle tree) and verification is performed
+    // on root stored in the index or value slot
+    // if it is not set verification is performed on according to the slotIndex. Value selected from the
+    // provided slot. For example if slotIndex is `1` value gets from `i_1` slot. If `4` from `v_1`.
+    signal output merklized;
+
+    // userID output signal will be assigned with ProfileID ProfileID(UserID, nonce),
```
[the remark this answers is not in the document]
Comment (demonsh) [reply circuits/offchain/credentialAtomicQuerySigOffChain.circom:43]: In this case it is not just a hash, it is actually constructing new profile from UserID and nonce. 
Full algorithm in ProfileID template.

[the remark this answers is not in the document]
Comment (AndriianChestnykh) [reply circuits/offchain/credentialAtomicQuerySigOffChain.circom:43]: `SelectProfile` template? If yes, then `SelectProfile(UserGenesisID, nonce)`

File: circuits/lib/query/credentialJsonLDAtomicQueryMTP.circom (hunk 1 above)
[the remark this answers is not in the document]
Comment (demonsh) [reply circuits/lib/query/credentialJsonLDAtomicQueryMTP.circom:25]: fix

File: circuits/offchain/credentialAtomicQuerySigOffChain.circom (hunk 2 above)
[the remark this answers is not in the document]
Comment (demonsh) [reply circuits/offchain/credentialAtomicQuerySigOffChain.circom]: fixed

**Review by AndriianChestnykh — APPROVED** [review]

**Pull request merged**